Disaster-Response-Platform/backend/Controllers/user_controller.py

```diff
@@ -0,0 +1,88 @@
+from fastapi import APIRouter, HTTPException, Response, Depends, status
+from fastapi.security import OAuth2PasswordBearer,OAuth2PasswordRequestForm
+from jose import JWTError, jwt
+from typing import Optional
+from datetime import datetime, timedelta
+from Services import authentication
+from fastapi.responses import JSONResponse
+
+import config
+from Database.mongo import MongoDB
+from Models import user_model
+
+
+router = APIRouter()
+db = MongoDB.getInstance()
+# Secret key to sign and verify the JWT token
+SECRET_KEY = config.SECRET_KEY
+ALGORITHM = "HS256"
+ACCESS_TOKEN_EXPIRE_MINUTES = 30
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")
+userDb = MongoDB.get_collection('authenticated_user')
+
+
+@router.post("/signup")
+async def signup(currentUser :user_model.RegisterUser):
+    try:
+         
+        if (userDb.find_one({"username":currentUser.username}) !=None) : #if there is a user already existed with current username
+            return HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail="Username already taken")
+        if (userDb.find_one({"email":currentUser.email}) !=None):
+            return HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail="Email already taken")
+        if (not authentication.is_valid_password(currentUser.password)) : #if username is not existed in db but password contains less than 8 characters
+            return HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail="Password cannot be less than 8 characters")
+        if (not authentication.is_valid_phone_number(currentUser.phone_number)): #if phone number is not valid
+            return HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail="Phone number format is wrong")
+        hash= authentication.get_password_hash(currentUser.password)
+       
+        insert_result = userDb.insert_one({
+            "username": currentUser.username,
+            "first_name": currentUser.first_name,
+            "last_name": currentUser.last_name,
+            "email": currentUser.email,
+            "phone_number": currentUser.phone_number,
+            "hashed_password": hash
+        })
+
+       #todo what should itreturn, my brain does not process anymore
+        return {"status": "OK"}
+  
+    except Exception as e:
+        # Handle any other unexpected exceptions here
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail="An unexpected error occurred")
+
+# Login route
+@router.post("/token", response_model=user_model.Token)
```

Comment:
can you change the name as sign-in :)

Reply:
Makes the most sense

Reply:
Now it's ok. It seems these changes are done.